_posts/2016-03-16-AngularTS.md

```diff
@@ -0,0 +1,413 @@
+
+To encapsulate our code, the module will transpile to an IIFE around our components.
+This will avoid global code which helps prevent variables and function declarations from living longer than expected in the global scope, which also helps avoid variable collisions.
+
+
+## Entity Class
+Now that TypeScript supports OO, we can analyse our business problem and define the business objects into entity classes.
+When you analyse and define these entities you can define which properties and methods each entity needs.
+If you have a couple entities, you can even establish a relationship.
+This will provide a clear view on what you want to achieve and have the possibility to create multiple instances of these classes. 
+When building an entity class you can optionally define an interface to show what’s the intent. 
+
+{% highlight javascript %}
+module JWorks {
+
+    export interface IEmployee {
+         username:string;
+         name:string;
+         eat(food:string):void
+    }
+
+    export class Employee implements IEmployee {
+ 
+          constructor(public username:string, public name:string){}
+
+          eat(food:string):void{
+             //implementation
+          }
+    }
+}
+
+{% endhighlight %}
+
+To use your entity class in a controller you have to define the export key.
+This will expose the class to other classes.
+When exporting the interface you will use it as a data type.
+
+### Class as property
+{% highlight javascript %}
+Employee :IEmployee;
+{% endhighlight %}
+
+### Instance of the class
+{% highlight javascript %}
+employee = new Employee();
+{% endhighlight %}
+
+### Access property
+{% highlight javascript %}
+employee.name
+{% endhighlight %}
+
+### Call methods
+{% highlight javascript %}
+employee.eat(eggs);
+{% endhighlight %}
+
+
+## Controllers
+As you know the controller defines the model to the view of your application, methods for every action your require and the scope where you hold a two way binding.
+Because TS offers an object oriented architecture, we can use classes and interfaces instead of functions.
+Interfaces are as in OO languages a contract you make and implementable by classes.
+When implemented, all methods and properties have to be used in the class.
+Classes declare and implement the properties and methods exposed to the view.
+Every class has his own constructor function, in this function we can declare default property values and other initialisation code. 
+
+### Controller Interface
+{% highlight javascript %}
+module Jworks {
+
+    interface IEmployeeController{
+        person:IPerson;
+        save(person:IPerson):void;
+    }
+}
+{% endhighlight %}
+
+The interface will show you the intent of our controller and declare the properties and methods that will be used.
+When you look at the syntax, you see that the properties are strong typed and the type is declared after the colon.
+If you have no aware of which type a property has, you can choose for the general type ‘any’.
```

Comment:
Change to:
If you **aren't aware of the type of a property, you can fall back to** the general type `any`.